testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -411,6 +411,36 @@ public void testHiveViewWithParametrizedTypes()
                 row("varchar"));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testHiveViewWithTextualTypes()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS hive_view_textual");
+        onHive().executeQuery("DROP TABLE IF EXISTS hive_table_textual");
+
+        // columns with `char` type have a fixed length between 1 and 255, columns with `varchar` type can have a length between 1 and 65535
+        onHive().executeQuery("CREATE TABLE hive_table_textual(a char(1), b char(255), c varchar(1), d varchar(65365), e string)");
+        onHive().executeQuery("CREATE VIEW hive_view_textual AS SELECT * FROM hive_table_textual");
+        onHive().executeQuery("INSERT INTO TABLE hive_table_textual VALUES ('a', 'rainy', 'i', 'calendar', 'Boston Red Sox')");
+
+        assertViewQuery(
+                "SELECT * FROM hive_view_textual",
+                queryAssert -> queryAssert.containsOnly(row("a", "rainy" + " ".repeat(250), "i", "calendar", "Boston Red Sox")));
+        assertViewQuery(
+                "SELECT a, d FROM hive_view_textual WHERE e = 'Boston Red Sox'",
+                queryAssert -> queryAssert.containsOnly(row("a", "calendar")));
+
+        // Trino columns type signature does not correspond one to one with the Hive view columns type in case of `varchar` and `string` columns
```

Comment:
prefix with "Currently,", since we it's not a by-design limitation, only as-a-matter-of-fact